agents/rust/mediator/mediation/src/router.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 Naian G.
+// SPDX-License-Identifier: Apache-2.0
+// use crate::routes::coordination::handle_coord;
+use std::sync::Arc;
+
+use axum::{
+    routing::{get, post},
+    Router,
+};
+
+use crate::{
+    routes::{
+        forward::handle_forward, hello_world, json, json::respond_message_json,
+        pickup::handle_pickup,
+    },
+    storage,
+};
+
+pub async fn create_router() -> Router {
+    // Initialize and get a storage struct that implements MediatorPersistence trait
+    // (this handles connection to storage backend)
+    let storage = storage::get_persistence().await;
+    Router::new()
+        .route(
+            "/",
+            get(hello_world::handle_get).post(hello_world::handle_get),
+        )
+        .route(
+            "/json",
+            get(json::echo_message_json).post(respond_message_json),
+        )
+        .route("/forward", post(handle_forward))
+        .route("/pickup", post(handle_pickup))
```

Comment:
these can be either:
- a single `didcomm` endpoint
- or 2 endpoints as you have it, but then I'd suggest `/didcomm` for 3rd party fw messages and `/didcomm-client` for didcomm messages from clients (eg connections, mediation coordition, pickup)

2 endpoints can be really good idea from oversight perspective, in logs it would be easier to distinguish between incoming 3rd party messages, client messages and plot stats and graphs out of that data